syntax = "proto3";

package ntf.pb;

message NtfTableCreateArg {
	uint32 dpid = 1;
	uint32 max_entries = 2;
}

message NtfTableDeleteArg {
	uint32 dpid = 1;
}

message UserCentricNetworkToken {
	uint32 app_id = 1;
	string encryption_key = 2;
	repeated uint64 blacklist = 3;
}

message NtfEntryCreateArg {
	uint32 dpid = 1;
	UserCentricNetworkToken token = 2;
	oneof set_dscp {
		uint32 dscp = 3;
	}
	oneof set_rule_id {
		uint32 rule_id = 4;
	}
}

message NtfEntryModifyArg {
	uint32 dpid = 1;
	UserCentricNetworkToken token = 2;
	oneof set_dscp {
		uint32 dscp = 3;
	}
	oneof set_rule_id {
		uint32 rule_id = 4;
	}
}

message NtfEntryDeleteArg {
	uint32 dpid = 1;
	uint32 app_id = 2;
}
